extractor/src/main/java/org/schabi/newpipe/extractor/localization/Localization.java

```diff
@@ -99,4 +98,30 @@ public int hashCode() {
         result = 31 * result + (countryCode != null ? countryCode.hashCode() : 0);
         return result;
     }
+
+    /**
+     * Converts a three letter language code (ISO 639-3) to a Locale
+     * in the limit of Java Locale class.
+     * It can also handle 639-2/T but may return a wrong locale.
+     * from https://stackoverflow.com/a/674122/12680950
+     * <p>
+     * It should not be used with 639-2/B, since java already handle it.
+     * (Just use new Locale("an ISO 639-2/B code")).
+     *
+     * @param code a three letter language code
+     * @return the Locale corresponding
+     */
+    public static Locale getLocaleFromThreeLetterCode(String code) {
+        String[] languages = Locale.getISOLanguages();
+        Map<String, Locale> localeMap = new HashMap<>(languages.length);
+        for (String language : languages) {
+            Locale locale = new Locale(language);
+            localeMap.put(locale.getISO3Language(), locale);
+        }
+        if (localeMap.containsKey(code)) {
+            return localeMap.get(code);
+        } else {
+            return new Locale(code.substring(0, 2));
+        }
```

Comment:
Remove this part, as it's technically incorrect. Just throw an exception if it's not in the `localeMap`.